.changeset/nine-donuts-confess.md

```diff
@@ -0,0 +1,9 @@
+---
+'astro': patch
+---
+
+Improve suppport for `import.meta.env`.
+
+Prior to this change, all variables defined in `.env` files had to include the `PUBLIC_` prefix, meaning that they could potentially be visible to the client if referenced. 
+
+Now, Astro includes _any_ referenced variables defined in `.env` files on `import.meta.env` during server-side rendering, but only referenced `PUBLIC_` variables on the client.
```

Comment:
I can make note of this in the blog post, and then will share both with @sarah11918 to help update docs for the new logic.